### PR TITLE
doc,bgpv1: More failure scenario and wording improvement

### PR DESCRIPTION
**Pull request by YutaroHayakawa**
Node failure scenario and working improvement for the existing doc.

```release-note
doc,bgpv1: More failure scenario and wording improvement
```

### CODE REVIEW

File: Documentation/network/bgp-control-plane-operation.rst [hunk 1]
```diff
@@ -68,6 +68,36 @@ forwarded to a different node than before, will be reset.
 
 .. _Resilient Hashing: https://www.juniper.net/documentation/us/en/software/junos/interfaces-ethernet-switches/topics/topic-map/switches-interface-resilient-hashing.html
 
+Node Down
+---------
+
+If the node goes down, the BGP session will be lost, and the Cilium datapath
```
Comment (harsimran-pabla) [inline Documentation/network/bgp-control-plane-operation.rst]:
```suggestion
If the node goes down, the BGP sessions from this node will be lost, and the Cilium datapath
```

File: Documentation/network/bgp-control-plane-operation.rst [hunk 2]
```diff
@@ -68,6 +68,36 @@ forwarded to a different node than before, will be reset.
...
+forwarding will be stopped. If the node is shut down gracefully with a proper
+BGP shutdown procedure or TCP connection close, the BGP peer either stops
```
Comment (harsimran-pabla) [inline Documentation/network/bgp-control-plane-operation.rst]: Should we document what is the `proper BGP shutdown procedure` or graceful node shutdown mechanism ?

File: Documentation/network/bgp-control-plane-operation.rst [hunk 3]
```diff
@@ -68,6 +68,36 @@ forwarded to a different node than before, will be reset.
...
+forwarding immediately or takes some time to stop forwarding traffic to the
+node depending on the Graceful Restart settings. Since the Cilium datapath is
+stopping, the traffic will be dropped during this period. This is especially
+problematic when you advertise the route to the Service with
```
Comment (harsimran-pabla) [inline Documentation/network/bgp-control-plane-operation.rst]: I think an explanation of why this is problematic is useful. 

BGP peer will continue to forward traffic to this node even if BGP session goes down due to node shutdown till GR restart timer expires.

File: Documentation/network/bgp-control-plane-operation.rst [hunk 4]
```diff
@@ -68,6 +68,36 @@ forwarded to a different node than before, will be reset.
...
+``externalTrafficPolicy=Cluster``.
+
+Mitigation
```
Comment (harsimran-pabla) [inline Documentation/network/bgp-control-plane-operation.rst:141]: Maybe we can have subsections under this topic. Voluntary node down, Involuntary node down.

File: Documentation/network/bgp-control-plane-operation.rst [hunk 5]
```diff
@@ -68,6 +68,36 @@ forwarded to a different node than before, will be reset.
...
+~~~~~~~~~~
+
+When a node is voluntarily shut down, by gracefully service-out the node, you
+can avoid the packet loss. For example, you can first drain the node to evict
+all workloads. After that, deconfigure the BGP session by removing the label
```
Comment (harsimran-pabla) [inline Documentation/network/bgp-control-plane-operation.rst]:
```suggestion
all workloads. After that, deconfigure the BGP sessions by removing CiliumBGPPeeringPolicy node selector label
```

File: Documentation/network/bgp-control-plane-operation.rst [hunk 6]
```diff
@@ -68,6 +68,36 @@ forwarded to a different node than before, will be reset.
...
+from the Node object to stop traffic towards the Services with
+``externalTrafficPolicy=Cluster``. Then, wait for a period of time until the
+BGP peer stops forwarding traffic to the node and finally shut down the node.
```
Comment (harsimran-pabla) [inline Documentation/network/bgp-control-plane-operation.rst]:
```suggestion
BGP peer removes routes towards the node and finally shut down the node.
```

File: Documentation/network/bgp-control-plane-operation.rst [hunk 7]
```diff
@@ -68,6 +68,36 @@ forwarded to a different node than before, will be reset.
...
+
+When a node is involuntarily shut down, there's no direct mitigation. You can
+choose to use or not to use the BGP Graceful Restart feature, depending on the
```
Comment (harsimran-pabla) [inline Documentation/network/bgp-control-plane-operation.rst]:
```suggestion
choose to not use the BGP Graceful Restart feature, depending on the
```

File: Documentation/network/bgp-control-plane-operation.rst [hunk 8]
```diff
@@ -68,6 +68,36 @@ forwarded to a different node than before, will be reset.
...
+trade-off between the failure detection time and stability. Disabling the
```
Comment (harsimran-pabla) [inline Documentation/network/bgp-control-plane-operation.rst]:
```suggestion
trade-off between the failure detection time vs stability provided by graceful restart in cases of Cilium pod restarts. Disabling the
```

File: Documentation/network/bgp-control-plane-operation.rst [hunk 9]
```diff
@@ -68,6 +68,36 @@ forwarded to a different node than before, will be reset.
...
+Graceful Restart allows the BGP peer to stop forwarding faster. Even if the
```
Comment (harsimran-pabla) [inline Documentation/network/bgp-control-plane-operation.rst]:
```suggestion
Graceful Restart allows the BGP peer to withdraw routes faster. Even if the
```

File: Documentation/network/bgp-control-plane-operation.rst [hunk 10]
```diff
@@ -68,6 +68,36 @@ forwarded to a different node than before, will be reset.
...
+node is shut down ungracefully, the worst case time to stop forwarding is the
```
Comment (harsimran-pabla) [inline Documentation/network/bgp-control-plane-operation.rst]:
```suggestion
node is shut down ungracefully, the worst case time for peer to withdraw routes is the
```

File: Documentation/network/bgp-control-plane-operation.rst [hunk 11]
```diff
@@ -68,6 +68,36 @@ forwarded to a different node than before, will be reset.
...
+BGP hold time. When the Graceful Restart is enabled, the BGP peer may need hold
+time + restart time to stop forwarding traffic to the node.
```
Comment (harsimran-pabla) [inline Documentation/network/bgp-control-plane-operation.rst]:
```suggestion
time + restart time to withdraw routes received from the node.
```

File: Documentation/network/bgp-control-plane-operation.rst (hunk 3 above)
Comment (learnitall) [inline Documentation/network/bgp-control-plane-operation.rst]: Grammar suggestions:

```suggestion
stopping, traffic will be dropped during this period. This is especially
problematic when you advertise the route to a Service with
```

File: Documentation/network/bgp-control-plane-operation.rst [hunk 12]
```diff
@@ -68,6 +68,36 @@ forwarded to a different node than before, will be reset.
...
+node depending on the Graceful Restart settings. Since the Cilium datapath is
+stopping, the traffic will be dropped during this period. This is especially
+problematic when you advertise the route to the Service with
+``externalTrafficPolicy=Cluster``.
+
+Mitigation
+~~~~~~~~~~
+
+When a node is voluntarily shut down, by gracefully service-out the node, you
+can avoid the packet loss. For example, you can first drain the node to evict
```
Comment (learnitall) [inline Documentation/network/bgp-control-plane-operation.rst]: Grammar suggestions:

```suggestion
When a node is voluntarily shut down, to gracefully service it out, you
can avoid packet loss. For example, you can first drain the node to evict
```

File: Documentation/network/bgp-control-plane-troubleshooting.rst [hunk 13]
```diff
@@ -53,20 +53,19 @@ established because there was a mismatch between the configured
 There could be various reasons why BGP peering is not established, such as a
 mismatch in BGP capability or an incorrect Peer IP address. BGP layer errors
 are likely to appear in the logs, but there are cases where low-level errors,
-such as lack of connectivity to the Peer IP or an eBGP Peer being ahead of the
-1-hop, may not be reflected in the logs. In such cases, using tools like
+such as lack of connectivity to the Peer IP or when eBGP peer is more than 1
```
Comment (learnitall) [inline Documentation/network/bgp-control-plane-troubleshooting.rst]:
```suggestion
such as lack of connectivity to the Peer IP or when an eBGP peer is more than 1
```

**Review by learnitall — CHANGES_REQUESTED** [review]
Great! Thank you for this. I have a couple of small grammar nits, otherwise LGTM.

File: Documentation/network/bgp-control-plane-operation.rst (hunk 12 above)
Comment (YutaroHayakawa) [reply Documentation/network/bgp-control-plane-operation.rst]: "by gracefully service it out" should be fine here. Users can avoid packet loss by gracefully shutting down the node.

**Review by learnitall — APPROVED** [review]
Thanks!

File: Documentation/network/bgp-control-plane-operation.rst (hunk 2 above)
Comment (YutaroHayakawa) [reply Documentation/network/bgp-control-plane-operation.rst]: I added a new `Node Shutdown` section. Please take a look.

File: Documentation/network/bgp-control-plane-operation.rst (hunk 3 above)
Comment (YutaroHayakawa) [reply Documentation/network/bgp-control-plane-operation.rst]: Made a `Involuntary Shutdown` section. I also added the `Voluntary Shutdown` section and referred to the `Node Shutdown` section from it.

File: Documentation/network/bgp-control-plane-operation.rst (hunk 4 above)
Comment (YutaroHayakawa) [reply Documentation/network/bgp-control-plane-operation.rst:141]: Made a Involuntary Shutdown section. I also added the Voluntary Shutdown section and referred to the Node Shutdown section from it.

File: Documentation/network/bgp-control-plane-operation.rst (hunk 3 above)
Comment (YutaroHayakawa) [reply Documentation/network/bgp-control-plane-operation.rst]: Yep, documented it.

**Review by harsimran-pabla — APPROVED** [review]
Thank you, looks great 🙏

**Pull request merged**